src/components/GraphUtils.js

```diff
@@ -0,0 +1,202 @@
+import * as d3 from 'd3';
+
+export function truncate(fullStr, strLen, separator='...') {
+  if (fullStr.length <= strLen) return fullStr;
+  
+  const sepLen = separator.length,
+    charsToShow = strLen - sepLen,
+    frontChars = Math.ceil(charsToShow / 2),
+    backChars = Math.floor(charsToShow / 2);
+  
+  return fullStr.substr(0, frontChars) +
+    separator +
+    fullStr.substr(fullStr.length - backChars);
+}
+
+export function clearClicked() {
+  d3.selectAll('.clicked').classed('clicked', false);
+}
+
+export function scaleGraph(svg, g, zoom) {
+  const width = svg.node().getBoundingClientRect().width;
+  const height = svg.node().getBoundingClientRect().height;
+
+  const graphX = g.node().getBBox().x;
+  const graphY = g.node().getBBox().y;
+  const graphWidth = g.node().getBBox().width;
+  const graphHeight = g.node().getBBox().height;
```

Comment:
If we decided to extract some functions into the visualization module, I suggest dropping the graph context, i.e., instead of using the `graph` keyword, use something generic, e.g., `bboxX`, `bboxWidth`, etc.